migrations/20241105164503_2.sql

```diff
@@ -0,0 +1,40 @@
+-- Force anything running this migration to use the right search path.
+set local search_path to underway;
+
+-- Remove attempt-specific columns from `task`
+alter table underway.task
+drop column if exists retry_count,
+drop column if exists initial_interval_ms,
+drop column if exists max_interval_ms,
+drop column if exists backoff_coefficient,
+drop column if exists error_message,
+drop column if exists started_at,
+drop column if exists succeeded_at,
+drop column if exists last_failed_at;
+
+alter table underway.task
+add column if not exists retry_policy jsonb not null default '{"max_attempts": 5, "initial_interval_ms": 1000, "max_interval_ms": 60000, "backoff_coefficient": 2.0}';
+
+alter table underway.task
+add column if not exists completed_at timestamp with time zone;
+
+alter table underway.task
+add column if not exists last_attempt_at timestamp with time zone;
+
+create table underway.task_attempt (
+    task_id         uuid not null,
+    task_queue_name text not null,
+    attempt_number  integer not null,
+
+    -- Task state.
+    state           underway.task_state not null default 'pending',
+
+    -- Error metatdata.
+    error_message   text,
```

Comment:
I considered adding a `error_details` JSONB column, but it's not clear what would live there. For example, `std::error::Error` is the limit of the API surface we have and at most it seems `sources` could be interest but it's unclear if that's actually valuable and could be added later.

Reply:
in clojure such things where simplier to handle cause one can just throw exception which contains any object
so it allows to have meta data on error itself (mostly related to domain specific errors, like `Order not in current state`)
Maybe indeed there is no much data to save

Reply:
I think e.g. `anyhow`, etc provide ways of enriching errors (e.g. they provide `context` methods) and that's potentially additional metadata that could be stored. 

However, I don't like the idea of using `anyhow` here: we have a concrete error type (`task::Error`) and so we can directly augment it if we want. That said, I don't know what additional metadata we'd want. Most of it will be provided by the `Display` impl I think and so `to_string` is already giving us that.

I'm certainly open to increasing the fidelity here, so if you have ideas about more context we could embed I'll be happy to explore that.

Reply:
it feels like it is nice to have more info about error, but i cant figure out what to put there